lib/awful/mouse/init.lua

```diff
@@ -326,6 +325,19 @@ end
 
 function mouse.object.set_current_widget() end
 
+--- Get the current widget geometry.
+-- @property current_widget_geometry
+-- @tparam ?table The geometry.
+-- @see current_widget
+
+function mouse.object.get_current_widget_geometry()
+    local _, ret = mouse.object.get_current_widget()
+
+    return ret
+end
+
+function mouse.object.set_current_widget_geometry() end
```

Comment:
Hm. Perhaps we should try getting rid of these "doing-nothing-setters" (and yes, this new idea doesn't belong into this PR).
What do you think about not allowing properties if there is a getter but not a setter in `gears.object.properties`?
Something like the following (untested):
```diff
diff --git a/lib/gears/object/properties.lua b/lib/gears/object/properties.lua
index 9b98868..a35239d 100644
--- a/lib/gears/object/properties.lua
+++ b/lib/gears/object/properties.lua
@@ -83,6 +83,12 @@ function object.capi_index_fallback(class, args)
             return args.setter_fallback(cobj, prop, value)
         end
 
+        -- If a getter exists but not a setter, then the property is read-only
+        if args.getterer_class and args.getter_class[getter_prefix..prop] then
+            -- error("Tried to set read-only-property (psychon always adds error()s, Elv13 doesn't like them")
+            return
+        end
+
         local fallback = properties[cobj] or cobj_register(cobj)
 
         -- Use the fallback property table
```